.github/workflows/publish.yml

```diff
@@ -0,0 +1,29 @@
+name: Publish NuGet Packages to nuget.org
+
+on:
+  release:
+    types: [published]
+
+jobs:
+  publish:
+    runs-on: ubuntu-latest
+    defaults:
+      run:
+        shell: pwsh
+    steps:
+      - name: Download NuGet Packages from Release
+        run: |
+          Write-Host "GITHUB_REF: '$($env:GITHUB_REF)'"
+          $prefix = "refs/tags/"
+          $name = $($env:GITHUB_REF).Substring($prefix.Length)
+          Write-Host "name: '$name'"
+          gh release download $name --repo $GITHUB_REPOSITORY --pattern '*.nupkg' --pattern '*.snupkg'
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+
+      - name: Publish to nuget.org
+        run: |
+          dotnet nuget push "*.nupkg" --api-key $env:NUGET_API_KEY --source https://api.nuget.org/v3/index.json
+        env:
+          NUGET_API_KEY: ${{ secrets.NUGET_API_KEY }}

```

Comment:
I might be mistaken, but don't you need to also publish the snupkg file as an explicit step?